src/backend/tests/integration/components/astra/test_astra_component.py

```diff
@@ -188,10 +197,7 @@ def test_astra_vectorize_with_provider_api_key():
             store.delete_collection()
 
 
-@pytest.mark.skipif(
-    not check_env_vars("ASTRA_DB_APPLICATION_TOKEN", "ASTRA_DB_API_ENDPOINT"),
-    reason="missing env vars",
-)
+@pytest.mark.api_key_required
```

Comment:
Is the split of `api_key_required` vs. `no_api_key_required` enough? (I think for now, yes). In the future, it would be nice if CI automatically ran specific integration tests if it detected changes to specific components (for people with access to API Keys, i.e. us).